pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -311,3 +305,21 @@ func (p *LaunchTemplateProvider) GetCABundle(ctx context.Context) (*string, erro
 	logging.FromContext(ctx).Debugf("Discovered caBundle, length %d", len(transportConfig.TLS.CAData))
 	return ptr.String(base64.StdEncoding.EncodeToString(transportConfig.TLS.CAData)), nil
 }
+
+func tagsFor(options *launchTemplateOptions) []*ec2.Tag {
+	var tags []*ec2.Tag
+	// Combine user defined and karpenter tags. If keys conflict, choose the latter.
```

Comment:
This is interesting. Should we validate user tags to ensure they can't collide? Seems like perhaps surprising behavior to have the user defined tags overridden.